frontend/app/routes/settings/change-password.tsx

```diff
@@ -0,0 +1,180 @@
+import { AlertCircle, LoaderIcon } from "lucide-react";
+import React from "react";
+import { redirect, useFetcher, useNavigation } from "react-router";
+import { toast } from "sonner";
+import { apiClient } from "~/api/client";
+import { Alert, AlertDescription, AlertTitle } from "~/components/ui/alert";
+import { Button, SubmitButton } from "~/components/ui/button";
+import {
+  FieldSet,
+  FieldSetLabel,
+  FieldSetPasswordToggleInput
+} from "~/components/ui/fieldset";
+import { Separator } from "~/components/ui/separator";
+import { userQueries } from "~/lib/queries";
+import {
+  type ErrorResponseFromAPI,
+  getFormErrorsFromResponseData
+} from "~/lib/utils";
+import { queryClient } from "~/root";
+import { getCsrfTokenHeader, metaTitle } from "~/utils";
+import type { Route } from "./+types/change-password";
+
+export const meta: Route.MetaFunction = () => [metaTitle("Account Settings")];
+
+export async function clientAction({ request }: Route.ClientActionArgs) {
+  const formData = await request.formData();
+  const intent = formData.get("intent")?.toString();
+
+  switch (intent) {
+    case "change_password":
+      return changePassword(formData);
+    default: {
+      throw new Error("Unexpected intent");
+    }
+  }
+}
+
+async function changePassword(formData: FormData) {
+  const credentials = {
+    current_password: formData.get("current_password")!.toString(),
+    new_password: formData.get("new_password")!.toString(),
+    confirm_password: formData.get("confirm_password")!.toString()
+  };
```

Comment:
```suggestion
  const credentials = {
    current_password: formData.get("current_password")!.toString(),
    new_password: formData.get("new_password")!.toString(),
    confirm_password: formData.get("confirm_password")!.toString()
  } satisfies RequestInput<'post',  "/api/auth/change-password/">;
```

No intent if we have only one form in the page.